org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend

```diff
@@ -46,7 +46,8 @@ class JavaDerivedStateComputer {
 			unloader.unloadRoot(eObject) 
 		}
 		resource.getContents().clear
-		
+		if (resource.isInfoFile)
```

Comment:
If I understand this correctly, JavaResources for info files do have a strong reference to a compilation unit.
Should that one be released, too? Assuming we would never have that reference, the null check below would have been sufficient. Other code here could be simplified, too.

Reply:
yes and no. i dont see where this reference would be installed in the first place.
or do you mean in other cases?

Reply:
I was thinking about a special treatment a little earlier in the lifecycle, e.g. in `JavaResource.doLoad(InputStream, Map<?, ?>)`

Reply:
i am not sure if that would have any unwanted side effects

Reply:
True. That's also the reason why I approved this minimally invasive fix here.